_ml-commons-plugin/agents-tools/tools/query-planning-tool.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: default
+title: Query Planning tool
+has_children: false
+has_toc: false
+nav_order: 50
+parent: Tools
+grand_parent: Agents and tools
+---
+
+<!-- vale off -->
+# Query Planning tool
+plugins.ml_commons.agentic_search_enabled: true
+{: .label .label-purple }
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, join the discussion on the [OpenSearch forum](https://forum.opensearch.org/).    
+{: .warning}
+<!-- vale on -->
+
+The `QueryPlanningTool` generates an OpenSearch Query DSL query string from a natural language question.
+
+## Step 1: Enable the agentic search feature flag
+
+To use the `QueryPlanningTool`, you must first enable the `agentic_search_enabled` setting:
+
+```json
+PUT _cluster/settings
+{
+  "persistent" : {
+    "plugins.ml_commons.agentic_search_enabled" : true
+  }
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch responds with an acknowledgment:
+
+```json
+{
+  "acknowledged": true,
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "agentic_search_enabled": "true"
+      }
+    }
+  },
+  "transient": {}
+}
+```
+
+## Step 2: Create a connector for a model
+
+The following example request creates a connector for a model hosted on Amazon Bedrock:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "Amazon Bedrock Claude 3.7-sonnet connector",
+    "description": "connector for base agent with tools",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "parameters": {
+      "region": "us-west-2",
+      "service_name": "bedrock",
+      "model": "us.anthropic.claude-3-7-sonnet-20250219-v1:0",
+      "system_prompt":"please help answer the user question "
+    },
+    "credential": {
+      "access_key": "<YOUR_ACCESS_KEY>",
+      "secret_key": "<YOUR_SECRET_KEY>",
+      "session_token": "<YOUR_SESSION_TOKEN>"
+    },
+    "actions": [
+      {
+        "action_type": "predict",
+        "method": "POST",
+        "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/converse",
+        "headers": {
+          "content-type": "application/json"
+        },
+        "request_body": "{ \"system\": [{\"text\": \"${parameters.system_prompt}\"}], \"messages\": [${parameters._chat_history:-}{\"role\":\"user\",\"content\":[{\"text\":\"${parameters.user_prompt}\"}]}${parameters._interactions:-}]${parameters.tool_configs:-} }"
+      }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch responds with a connector ID:
+
+```json
+{
+  "connector_id": "NtjQi5gBOh0h20Y9GBWK"
+}
+```
+
+## Step 3: Register and deploy the model
```

Comment:
Should we merge the deploy and create connector in the same step to avoid one more step for the setup?

Reply:
merged in inline creation of model register and deploy